libs/libc/misc/lib_utsname.c

```diff
@@ -93,7 +93,8 @@ int uname(FAR struct utsname *name)
 
   strlcpy(name->release,  CONFIG_VERSION_STRING, sizeof(name->release));
 
-#if defined(__DATE__) && defined(__TIME__)
+#if defined(__DATE__) && defined(__TIME__) && \
+    !defined(CONFIG_LIBC_UNAME_DISABLE_TIMESTAMP)
```

Comment:
it's better to guard .PHONY target to avoid the unnecessary compiling

Reply:
Hi @xiaoxiang781216 as I commented in the previous PR, doing it there is not ideal, if you run "make clean" and compile again it will generate a new type of binary

Reply:
Yes, I know. My suggestion is adding the check in both place.

Reply:
Ah ok

Reply:
Done